checker/src/main/java/org/checkerframework/checker/signature/SignatureAnnotatedTypeFactory.java

```diff
@@ -157,6 +157,9 @@ public Void visitMethodInvocation(MethodInvocationTree tree, AnnotatedTypeMirror
                 } else if ((oldChar == '/' && newChar == '.')
                         && receiverType.getAnnotation(InternalForm.class) != null) {
                     type.replaceAnnotation(BINARY_NAME);
+                } else if ((oldChar == '$' && newChar == '.')
+                        && receiverType.getAnnotation(BinaryName.class) != null) {
+                    type.replaceAnnotation(DOT_SEPARATED_IDENTIFIERS);
```

Comment:
This is incorrect.  "MyClass$22" is a binary name, but "MyClass.22" is not a dot-separated identifier.
Please remove this conversion rule.

(Actually, this is a bug that I discovered in code review and fixed in commit 084e251629.   For efficiency, I don't point out every issue in the code to you.)